src/hooks/stores/usePaginatedDataStore.ts

```diff
@@ -0,0 +1,97 @@
+import { useState, useContext, useEffect } from 'react';
+import { ApiClientsContext } from 'pages/_app';
+import {
+  useGetAllGrantsLazyQuery,
+  GetAllGrantsQuery,
+} from 'src/generated/graphql';
+import { useAccount } from 'wagmi';
+import { useToast } from '@chakra-ui/react';
+
+export type PaginatedData<T> = {
+  items: T[]
+  loading: boolean
+  nextPage?: string
+  hasMore: boolean
+  error?: Error
+};
+
+function usePaginatedDataStore<T>() {
+  const PAGE_SIZE = 40;
+  const [currentPage, setCurrentPage] = useState(0);
+  const [grants, setGrants] = useState<GetAllGrantsQuery['grants']>([]);
```

Comment:
should be generic so we can run different kinds of paginated queries -- ideally this code can be used everywhere we need paginated data